packages/screens/Launchpad/components/UriTab.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import { useForm } from "react-hook-form";
+import { View } from "react-native";
+
+import { BrandText } from "../../../components/BrandText";
+import { TextInputCustom } from "../../../components/inputs/TextInputCustom";
+import { SpacerColumn } from "../../../components/spacer";
+import { neutral00, neutral77 } from "../../../utils/style/colors";
+import { fontSemibold14 } from "../../../utils/style/fonts";
+import { layout } from "../../../utils/style/layout";
+import { ExistingBaseUrlValues } from "../CreateCollection.type";
+
+export const UriTab: React.FC = () => {
+  const { control } = useForm<ExistingBaseUrlValues>({
+    defaultValues: {
+      baseTokenUri: "",
+      coverImageUrl: "",
+    },
+    mode: "onBlur",
+  });
+
+  return (
+    <View style={{ width: "100%", flex: 1 }}>
+      <View
+        style={{
+          flexDirection: "row",
+          justifyContent: "center",
+        }}
+      >
+        <View
+          style={{
+            width: "100%",
+            maxWidth: 416,
+            margin: layout.spacing_x2,
+          }}
+        >
+          <BrandText
+            style={[fontSemibold14, { color: neutral77, textAlign: "center" }]}
+          >
+            Though Teritori's tr721 contract allows for off-chain metadata
+            storage, it is recommended to use a decentralized storage solution,
+            such as IPFS. You may head over to NFT.Storage and upload your
+            assets & metadata manually to get a base URI for your collection.
+          </BrandText>
+
+          <SpacerColumn size={2} />
+
+          <View>
+            <TextInputCustom<ExistingBaseUrlValues>
```

Comment:
I didn't put comments on all of this variations but was it not better to create a `TextInput` component for each type ? Because each type you use with `TextInputCustom` have their props, but it will be more maintainable and readable if you create a component for each type no ? (It's a proposal)
For me `TextInputCustom` have to be used for exceptions, or component that will be used only once 👍

Reply:
Can you please share some more details? Right now, it is quite confusing.

Reply:
Right now you use `TextInputCustom` with a specific type (here `ExistingBaseUrlValues`) but actually we try to maximize specific components to be more maintainable and readable 👍 
So for this one you can create a component `TextInputLaunchpadUrlValues` (or a name like that) specific to that and you'll call it everywhere you want to use it 👍 
I think you can do that only for components that you reuse many times
Is it more clear ?

Reply:
Thanks. Let me do something more readable.

Reply:
@clegirar I updated it.
Please check and let me know if you have any suggestions.

Reply:
I think i liked it 👍 The main issue is that is not the main purpose of this PR but i think we can keep it, because it's fine